### DISCUSSION TITLE
Not able to resolve webpack loader issue

### DISCUSSION THREAD

**Discussion by Madavic51** [webpack/webpack#18757] [Q&A] [answered]
Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.module.rules[1] should be one of these:
   ["..." | false | 0 | "" | null | undefined | object { assert?, compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, issuerLayer?, layer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, scheme?, sideEffects?, test?, type?, use?, with? }, ...]
   -> A rule.
   Details:
    * configuration.module.rules[1].loader should be a non-empty string.

**Comment by alexander-akait** [comment]
please show your configuration

**Reply by Madavic51** [reply]
module: {
    rules: [{ // handles dev component .scss from /app 
        test: /\.scss$/,
        
        include: [
            path.resolve(__dirname, 'test')
        ],
        loader: ['to-string-loader', 'css-loader', 'sass-loader']
    },

**Reply by alexander-akait** [reply, marked as answer]
please use `use` keyword, as here - https://github.com/webpack-contrib/css-loader?tab=readme-ov-file#getting-started)

**Reply by Madavic51** [reply]
Getting different error [webpack-cli] Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'inline'. These properties are valid:
   object { allowedHosts?, bonjour?, client?, compress?, devMiddleware?, headers?, historyApiFallback?, host?, hot?, ipc?, liveReload?, onListening?, open?, port?, proxy?, server?, app?, setupExitSignals?, setupMiddlewares?, static?, watchFiles?, webSocketServer? }

**Reply by Madavic51** [reply]
const { merge } = require('webpack-merge');
const baseConfig = require('./webpack.config.js');
const path = require('path');
module.exports = merge(baseConfig, {
    mode: 'development',
    module: {
        rules: [{ // handles dev component .scss from /app 
            test: /\.scss$/,
            
            include: [
                path.resolve(__dirname, '/app')
            ],
            use: ['to-string-loader', 'css-loader', 'sass-loader']
        },

        { // handles dev global .scss from /styles - inlines to the page
            test: /\.scss$/,
            include: path.resolve(__dirname, '/styles'),
            use: ['style-loader', 'css-loader', 'sass-loader']
        },
        ]
    },
    devtool: 'inline-cheap-module-source-map', // Useful for debugging
   
        
    devServer: {
                port: 9999,
                contentBase: path.resolve(__dirname, 'wwwroot'),
                publicPath: '/',
                inline: true,
                compress: true,
                historyApiFallback: true,
                proxy: [{
                    context: ['url']
                    target: 'https://localhost',
                    changeOrigin: true
                },
                {
                    context: ['/config', '/testt1'],
                    target: 'http://localhost:8888'
                }
                   
                ]
            },

    
})

**Reply by alexander-akait** [reply]
We don't have `inline` for `devServer`, just remove it

**Reply by Madavic51** [reply]
still same after removing

**Reply by alexander-akait** [reply]
it means you still have this option, double check it

**Reply by Madavic51** [reply]
[webpack-cli] Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'contentBase'. These properties are valid:
   object { allowedHosts?, bonjour?, client?, compress?, devMiddleware?, headers?, historyApiFallback?, host?, hot?, ipc?, liveReload?, onListening?, open?, port?, proxy?, server?, app?, setupExitSignals?, setupMiddlewares?, static?, watchFiles?, webSocketServer? }

for the above config which i sent previously getting this error 
if i remove content which is the alternative property to use
If i comment that to match  with latest version webpack i below error 
   no extension
        Field 'browser' doesn't contain a valid alias configuration

**Reply by alexander-akait** [reply]
Please follow the error message and check with the current documentation.

`contentBase` was removed a long time ago in favor of https://webpack.js.org/configuration/dev-server/#devserverstatic

**Reply by Madavic51** [reply]
Thanks 

is this format correct ? with latest webpack proxy: [{
                    context: ['url']
                    target: 'https://localhost',
                    changeOrigin: true
                },
                {
                    context: ['/config', '/testt1'],
                    target: 'http://localhost:8888'
                }

**Reply by Madavic51** [reply]
And inside  plugins: [
     new CleanWebpackPlugin(),
     
     with webpack 4 i was passing values  new CleanWebpackPlugin(['wwwroot/dist', 'wwwroot/index.html', 'wwwroot/silent-renew.html']),
     
     now with webpack5 i wont accept how to pass this values

**Reply by alexander-akait** [reply]
Should work, but need test in your env

**Reply by Madavic51** [reply]
Thanks
for the previous question am using  CleanWebpackPlugin  in previous webpack 4 i had a parameters to pass now it wont accept if i pass i get below error how can wepass parameters


[webpack-cli] Error: clean-webpack-plugin only accepts an options object. See:

 i got answer for many of my questions one last question after resolving some build issues getting this

 ERROR in main
Module not found: Error: Can't resolve './src/index.js'

**Reply by alexander-akait** [reply]
Check your the `entry` option

**Reply by Madavic51** [reply]
what should i change here

const isDevBuild = process.argv.indexOf('--env.prod') < 0;
const path = require('path');
const webpack = require('webpack');

const { CleanWebpackPlugin } = require('clean-webpack-plugin');

console.log('webpack running: creating vendor dll...')

module.exports = {
    mode: isDevBuild ? 'development' : 'production',

    devtool: isDevBuild ? 'cheap-module-source-map' : '',

    resolve: {
        extensions: ['.js'],
        modules: ['node_modules']
    },

    entry: {
        vendor: [
            '@agm/core',
            '@angular/animations',
            '@angular/cdk',
            '@angular/common',
            '@angular/compiler',
            '@angular/core',
            '@angular/forms',
            '@angular/material',
            '@angular/material-moment-adapter',
            '@angular/platform-browser',
            '@angular/platform-browser-dynamic',
            '@angular/router',
            '@microsoft/signalr',
            '@ncstate/sat-popover',
            'angular-feather',
            'angular2-text-mask',
            'applicationinsights-js',
            'bootstrap',
            'canvas-to-blob',
            'core-js',
            'event-source-polyfill',
            'moment',
            'ng2-ckeditor',
            'ng2-dnd',
            'oidc-client',
            'parse-address-string',
            'popper.js',
            'reflect-metadata',
            'rxjs',
            'text-mask-addons',
            'zone.js'
        ]
    },

    output: {
        path: path.resolve(__dirname, 'wwwroot', 'dist'),
        filename: '[name].dll.js',
        library: '[name][hash]'
    },

    module: {
        rules: [{
            test: /\.(png|woff|woff2|eot|ttf|svg)(\?|$)/,
            include: path.resolve(__dirname, 'node_modules'),
             loader: 'url-loader',
            options: {
                limit: 100000,
                name: 'assets/img/[name].[ext]'
            }
        }]
    },

    plugins: [
        new CleanWebpackPlugin(['wwwroot/dist', 'wwwroot/index.html', 'wwwroot/silent-renew.html']),
        new webpack.DllPlugin({
            name: '[name][hash]',
            path: path.resolve(__dirname, 'wwwroot', 'dist', '[name].manifest.json'),
            context: path.resolve(__dirname, 'wwwroot', 'dist')
        })
    ]
}

**Reply by alexander-akait** [reply]
You need to set https://webpack.js.org/concepts/entry-points/

**Reply by Madavic51** [reply]
The above codewas from webpack.config.vendor.js
in wenpack.config.js i have like this 

onst path = require('path');
const webpack = require('webpack');

const HtmlWebpackPlugin = require('html-webpack-plugin');
const HtmlWebpackIncludeAssetsPlugin = require('html-webpack-include-assets-plugin');

// Common Webpack Configuration is stored 
const commonWebpackConfig = {
    resolve: {
        extensions: ['.ts', '.js'],
        modules: ['ClientApp', 'node_modules']
    },

    entry: {
        app: './ClientApp/boot.ts',
        silentRenew: './ClientApp/assets/silent-renew/silent-renew.ts'
    },

    module: {
        rules: [{ // handles typescript 
                test: /^(?!.*\.spec\.ts$).*\.ts$/,
                include: path.resolve(__dirname, 'ClientApp'),
                use: [
                     'ts-loader' ,'angular-router-loader' ,'angular2-template-loader' 
                ]
            },

            { // handles html
                test: /\.html$/,
                include: [
                    path.resolve(__dirname, 'ClientApp', 'app')
                ],
                use: ['raw-loader']
            }
        ]
    },

    plugins: [
        new webpack.EnvironmentPlugin({
            NODE_ENV: isDevBuild ? 'development' : 'production'
        }),
        new webpack.DllReferencePlugin({
            context: path.resolve(__dirname, 'wwwroot', 'dist'),
            manifest: require(path.resolve(__dirname, 'wwwroot', 'dist', 'vendor.manifest.json')),
        }),
        new HtmlWebpackPlugin({
            cache: true, // Emit the file only if it was changed
            chunks: ['app'], // only bring in app chunk to this file (see entry names)
            template: path.resolve(__dirname, `ClientApp/${indexEjsFile}`),
            filename: path.resolve(__dirname, 'wwwroot/index.html'),
        }),
        new HtmlWebpackPlugin({
            cache: true, // Emit the file only if it was changed
            chunks: ['silentRenew'], // only bring in silentRenew chunk to this file (see entry names)
            template: path.resolve(__dirname, 'ClientApp/assets/silent-renew/silent-renew.ejs'),
            filename: path.resolve(__dirname, 'wwwroot/silent-renew.html'),
        }),
        new HtmlWebpackIncludeAssetsPlugin({
            assets: ['/dist/vendor.dll.js'],
            publicPath: false, // doesn't prefix 'assets' with anything
            append: false, // includes file above (prepends) other files
            hash: !isDevBuild
        })
    ]
}

module.exports = isDevBuild ? require('./webpack.dev.js') : require('./webpack.prod.js');

**Reply by alexander-akait** [reply]
If you need full support - please create reprducible test repo using github, thank you
